In [2]:
import pandas as pd
import pandas as pd
from sqlalchemy import create_engine
%load_ext sql

In [3]:
# Create a connection to the PostgreSQL database
engine=create_engine('postgresql+psycopg2://postgres:password@localhost:5432/medsupply')
connection=engine.connect()
print("Connected:", connection.closed == False)

Connected: True


In [4]:
%sql postgresql+psycopg2://postgres:password@localhost:5432/medsupply

'Connected: postgres@medsupply'

### Percentage Revenue for each Invoice status

In [5]:

%%sql

WITH revenue_cte AS (
    SELECT 
        SUM(CASE 
            WHEN tm.invoice_status = 'Completed' THEN tm.grand_total
            ELSE 0 
            END)
            AS completed_revenue,
        SUM(tm.grand_total) AS total_revenue
    FROM transactions_main AS tm
)
SELECT 
    completed_revenue,
    total_revenue,
    ROUND((completed_revenue * 100.0 / total_revenue)::NUMERIC, 2) AS completed_revenue_percentage
FROM revenue_cte;


 * postgresql+psycopg2://postgres:***@localhost:5432/medsupply
1 rows affected.


[(627238579120.0, 1115631508288.0, Decimal('56.22'))]

dari total revenue 1115631508288, hampir 56.22% invoice yang sudah bisa di cairkan (complete)

In [6]:

%%sql

WITH revenue_cte AS (
    SELECT 
        SUM(CASE 
            WHEN tm.invoice_status = 'Sent to Tax' THEN tm.grand_total
            ELSE 0 
            END)
            AS sent_to_tax_revenue,
        SUM(tm.grand_total) AS total_revenue
    FROM transactions_main AS tm
)
SELECT 
    sent_to_tax_revenue,
    total_revenue,
    ROUND((sent_to_tax_revenue * 100.0 / total_revenue)::NUMERIC, 2) AS sent_to_tax_revenue_percentage
FROM revenue_cte;


 * postgresql+psycopg2://postgres:***@localhost:5432/medsupply
1 rows affected.


[(124862137686.0, 1115631508288.0, Decimal('11.19'))]

sementara ada 11.19% yang masih dalam proses naik pajak dan belum di lakukan tukar faktur/pencairan dana dengan customer

In [7]:
%%sql

WITH revenue_cte AS (
    SELECT 
        SUM(CASE 
            WHEN tm.invoice_status = 'Awaiting DN Signed' THEN tm.grand_total
            ELSE 0 
            END)
            AS awaiting_dn_signed_revenue,
        SUM(tm.grand_total) AS total_revenue
    FROM transactions_main AS tm
)
SELECT 
    awaiting_dn_signed_revenue,
    total_revenue,
    ROUND((awaiting_dn_signed_revenue * 100.0 / total_revenue)::NUMERIC, 2) AS awaiting_dn_signed_revenue_percentage
FROM revenue_cte;

 * postgresql+psycopg2://postgres:***@localhost:5432/medsupply
1 rows affected.


[(188552217116.0, 1115631508288.0, Decimal('16.90'))]

ada hampir 17% transaksi yang belum bisa naik pajak karena menunggu DN dari rumah sakit/klinik setelah barang di kirim

In [8]:
%%sql

WITH revenue_cte AS (
    SELECT 
        SUM(CASE 
            WHEN tm.invoice_status = 'Customer Confirmation' THEN tm.grand_total
            ELSE 0 
            END)
            AS customer_confirmation_revenue,
        SUM(tm.grand_total) AS total_revenue
    FROM transactions_main AS tm
)
SELECT 
    customer_confirmation_revenue,
    total_revenue,
    ROUND((customer_confirmation_revenue * 100.0 / total_revenue)::NUMERIC, 2) AS customer_confirmation_revenue_percentage
FROM revenue_cte;

 * postgresql+psycopg2://postgres:***@localhost:5432/medsupply
1 rows affected.


[(107861812938.0, 1115631508288.0, Decimal('9.67'))]

ada transaksi yang menunggu confirmasi dari customer karena barang dari PO customer ada sebagian inden sebanyak 9.67%

In [9]:
%%sql

WITH revenue_cte AS (
    SELECT 
        SUM(CASE 
            WHEN tm.invoice_status IS NULL THEN tm.grand_total
            ELSE 0 
            END)
            AS completed_revenue,
        SUM(tm.grand_total) AS total_revenue
    FROM transactions_main AS tm
)
SELECT 
    completed_revenue,
    total_revenue,
    ROUND((completed_revenue * 100.0 / total_revenue)::NUMERIC, 2) AS completed_revenue_percentage
FROM revenue_cte;

 * postgresql+psycopg2://postgres:***@localhost:5432/medsupply
1 rows affected.


[(53558536738.0, 1115631508288.0, Decimal('4.80'))]

4.8% transaksi masih menggantung statusnya dan ini lumayan besar dan harus di telusuri

In [10]:
%%sql

WITH revenue_cte AS (
    SELECT 
        SUM(CASE 
            WHEN tm.invoice_status = 'Cancelled' THEN tm.grand_total
            ELSE 0 
            END)
            AS completed_revenue,
        SUM(tm.grand_total) AS total_revenue
    FROM transactions_main AS tm
)
SELECT 
    completed_revenue,
    total_revenue,
    ROUND((completed_revenue * 100.0 / total_revenue)::NUMERIC, 2) AS completed_revenue_percentage
FROM revenue_cte;

 * postgresql+psycopg2://postgres:***@localhost:5432/medsupply
1 rows affected.


[(13558224690.0, 1115631508288.0, Decimal('1.22'))]

kurang dari 2% setelah confirmasi dari customer, transaksi di cancel terkait barang yang inden